### PR TITLE
Fix data load for user with id 0

### PR DESCRIPTION
**Pull request by mhgtlc**
Fix null data for user with id 0, `\Pimcore\Model\User::getById(0);` now returns null

### CODE REVIEW

File: models/User/AbstractUser/Dao.php [hunk 1]
```diff
@@ -40,7 +40,7 @@ public function getById($id)
             $data = $this->db->fetchRow('SELECT * FROM users WHERE `id` = ?', $id);
         }
 
-        if (!empty($data['id'])) {
+        if (!empty($data)) {
```
Comment (brusch) [inline models/User/AbstractUser/Dao.php]:
```suggestion
        if ($data) {
```

File: models/User/AbstractUser/Dao.php [hunk 2]
```diff
@@ -56,7 +56,7 @@ public function getByName($name)
     {
         $data = $this->db->fetchRow('SELECT * FROM users WHERE `type` = ? AND `name` = ?', [$this->model->getType(), $name]);
 
-        if (!empty($data['id'])) {
+        if (!empty($data)) {
```
Comment (brusch) [inline models/User/AbstractUser/Dao.php]:
```suggestion
        if ($data) {
```

**Review by brusch — CHANGES_REQUESTED** [review]
Just a simple `if` should be sufficient 😉

File: models/User/AbstractUser/Dao.php (hunk 1 above)
Comment (mhgtlc) [reply models/User/AbstractUser/Dao.php]: Fixed

**Comment by brusch** [comment]
@mhgtlc could you please also sign the CLA? Thanks again!

**Comment by brusch** [comment]
@mhgtlc please see also the instructions above, you have to ensure the email address used for the Git commit is also mapped in your GitHub settings. 
https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user

You can check the used email address in the patch file: 
https://patch-diff.githubusercontent.com/raw/pimcore/pimcore/pull/10783.patch

it's located in the second line.

**Comment by mhgtlc** [comment]
> https://patch-diff.githubusercontent.com/raw/pimcore/pimcore/pull/10783.patch

Fixed, can you please check once again? Thank you

**Review by brusch — APPROVED** [review]

**Comment by brusch** [comment]
Thank you! 👍

**Pull request merged**